flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesPodTemplateTestUtils.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.kubernetes;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.PodSpec;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.fabric8.kubernetes.api.model.VolumeMountBuilder;
+
+import java.net.URL;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.assertThat;
+
+/** Utilities for the Kubernetes pod template tests. */
+public class KubernetesPodTemplateTestUtils {
+
+    public static final String TESTING_MAIN_CONTAINER_NAME = "testing-main-container";
+    public static final String TESTING_SIDE_CAR_CONTAINER_NAME = "sidecar-log-collector";
+
+    private static final String TESTING_TEMPLATE_FILE_NAME = "testing-pod-template.yaml";
+
+    public static String getPodTemplatePath() {
+        final URL podTemplateUrl =
+                KubernetesPodTemplateTestUtils.class
+                        .getClassLoader()
+                        .getResource(TESTING_TEMPLATE_FILE_NAME);
+        assertThat(podTemplateUrl, not(nullValue()));
+        return podTemplateUrl.getPath();
+    }
+
+    public static Container getInitContainerInPodTemplate() {
```

Comment:
Maybe add a comment that these are the containers specified in the `test-resources/testing-pod-template.yaml`. Maybe this should go to the class JavaDocs.